File: InventoryApp.Angular/src/app/models/book.ts
/* Defines the book entity */
export interface Book {
    id: number;
    bookTitle: string;
    topic: string;
    ISBN: string;
    tags?: string[];
    publishedDate: string;
    price: number;
    author: string;
    description: string;
    starRating: number;
    bookImageUrl: string;
}

export interface BookResolved {
    book: Book;
    error?: any;
}

File: DatingApp.API/Models/Book.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DatingApp.API.Models
{
    public class Book
    {
        public int Id { get; set; }

        public string BookTitle { get; set; }

        public string Topic { get; set; }

        public string ISBN { get; set; }

        public ICollection<BookTag> BookTags { get; set; }

        [DisplayFormat(DataFormatString="{0:d}")]
        public DateTime PublishedDate { get; set; }

        public float Price { get; set; }

        public string Author { get; set; }

        public string Description { get; set; }

        public float StarRating { get; set; }

        public string BookImageUrl { get; set; }
    }
}